frontend/src/static/js/v_authorship.js

```diff
@@ -245,20 +246,25 @@ window.vAuthorship = {
             ? this.getContributionFromAllAuthors(contributionMap)
             : contributionMap[this.info.author];
```

Comment:
can check valid file first then calculate linecnt

Reply:
Thanks for the suggestion. I tried to refactor it but I think the original intention is that file that is not binary and has 0 line count will not be included as valid files, so it may be necessary to compute the line count before checking whether the file is valid. Still, please let me know if there is a workaround as I am still learning this section

Reply:
Using line count is just some hacking when the group is merged since we did not have this binary type which always has 0 line count. It is ok to compute the lineCount first, but then your isValidFile logic has overlapping part with the lineCount section. You may want to read `authorship.json` to better understand the logic here.

Reply:
I have moved the validation to the position before the calculation of line count. However, I am still not sure about whether the changes made correspond to the given suggestion. Please let me know if the changes made are still inconsistent with the suggestion given and I will look into it again.

Reply:
Following your current implementation, the logic here should be

1. if not merged group -> check if in contribution map
2. if merged group -> check for all authors in the group, if > 0 summed contribution
3.  if satisfy either 1 or 2 -> a valid file for consideration, do the following computation

Also, don't use return within ForEach, do a filter first

Reply:
Sure, no problem